### PR TITLE
Various client api tweaks

### PR DESCRIPTION
**Pull request by bobbyrullo**
* change generateClientCreds to addClientCreds - more accurate reflection of the API
* add test to make sure that clients in sample file are valid, and that they can be authenticated against (this makes me worry less about screwing up the encoding/hashing of client_secrets)
* in preparation for removing ClientManagerFromClients, fix up a lot of copy-pasta in server tests and use testutil code instead to create fixtures
* finally, remove ClientManagerFromClients, in favor of a similar method in the repo layer

### CODE REVIEW

File: client/manager/manager.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ func NewClientManager(clientRepo client.ClientRepo, txnFactory repo.TransactionF
 	}
 }
 
-func NewClientManagerFromClients(clientRepo client.ClientRepo, txnFactory repo.TransactionFactory, clients []client.Client, options ManagerOptions) (*ClientManager, error) {
+func NewClientManagerFromClientsX(clientRepo client.ClientRepo, txnFactory repo.TransactionFactory, clients []client.Client, options ManagerOptions) (*ClientManager, error) {
```
Comment (ericchiang) [inline client/manager/manager.go]: ClientsX?

**Comment by ecordell** [comment]
LGTM

File: client/manager/manager.go (hunk 1 above)
Comment (ecordell) [inline client/manager/manager.go]: I assumed that was because this will be removed?

Comment (ericchiang) [inline client/manager/manager.go]: Hmm, maybe just a comment?

```
// DEPRICATED: Use xyz instead.
func NewClientManagerFromClients(...)
```

Comment (bobbyrullo) [inline client/manager/manager.go]: oops. forgot to delete.

**Comment by bobbyrullo** [comment]
Thanks for the review Evan!

**Pull request merged**